magicctapipe/io/io.py

```diff
@@ -528,110 +710,565 @@ def load_train_data_files(
-        sim_config["num_showers"][0]
-        * sim_config["shower_reuse"][0]
-        * len(np.unique(event_table["obs_id"]))
-    )
+def load_train_data_files_tel(input_dir, config, offaxis_min=None, offaxis_max=None, true_event_class=None):
+    """
+    Loads DL1-stereo data files and separates the shower events per
+    telescope combination type for training RFs.
 
-    min_viewcone_radius = sim_config["min_viewcone_radius"][0] * u.deg
-    max_viewcone_radius = sim_config["max_viewcone_radius"][0] * u.deg
+    Parameters
+    ----------
+    input_dir: str
+        Path to a directory where input DL1-stereo files are stored
+    config: dict 
+        yaml file with information about the telescope IDs. Typically called "config_general.yaml"
+    offaxis_min: str
+        Minimum shower off-axis angle allowed, whose format should be
+        acceptable by `astropy.units.quantity.Quantity`
+    offaxis_max: str
+        Maximum shower off-axis angle allowed, whose format should be
+        acceptable by `astropy.units.quantity.Quantity`
+    true_event_class: int
+        True event class of the input events
+    
+
+    Returns
+    -------
+    data_train: dict
+        Data frames of the shower events separated telescope-wise
+
+
+    Raises
+    ------
+    FileNotFoundError
+        If any DL1-stereo data files are not found in the input
+        directory
+    """
+    
+    TEL_NAMES, _ = telescope_combinations(config)
+    
+    # Find the input files
+    file_mask = f"{input_dir}/dl1_stereo_*.h5"
+
+    input_files = glob.glob(file_mask)
+    input_files.sort()
+
+    if len(input_files) == 0:
+        raise FileNotFoundError(
+            "Could not find any DL1-stereo data files in the input directory."
+        )
+
+    # Load the input files
+    logger.info("\nThe following DL1-stereo data files are found:")
+
+    data_list = []
+
+    for input_file in input_files:
+        logger.info(input_file)
+
+        df_events = pd.read_hdf(input_file, key="events/parameters")
+        data_list.append(df_events)
+
+    event_data = pd.concat(data_list)
+    event_data.set_index(GROUP_INDEX_TRAIN, inplace=True)
+    event_data.sort_index(inplace=True)
+
+    if offaxis_min is not None:
+        offaxis_min = u.Quantity(offaxis_min).to_value("deg")
+        event_data.query(f"off_axis >= {offaxis_min}", inplace=True)
+
+    if offaxis_max is not None:
+        offaxis_max = u.Quantity(offaxis_max).to_value("deg")
+        event_data.query(f"off_axis <= {offaxis_max}", inplace=True)
+
+    if true_event_class is not None:
+        event_data["true_event_class"] = true_event_class
+
+    event_data = get_stereo_events(event_data, config, group_index=GROUP_INDEX_TRAIN)
+
+    data_train = {}
+
+    # Loop over every telescope
+    for tel_id in TEL_NAMES.keys():
+        df_events = event_data.query(f"tel_id == {tel_id}")
+
+        if not df_events.empty:
+            data_train[tel_id] = df_events
+
+    return data_train
+
+'''
```

Comment:
if not used anymore, simply remove the lines instead of having 200 lines of commented code! This holds even if you think you need these lines at some point in the future. Git keeps all history, so you can simply check what was there when the current changes were not applied.

Reply:
Yes, as in the comment above, we will remove them (they are needed in the next PR, so we can remove them now)

Reply:
ok, now I see. Well, then you can keep them. It is just for me (and anyone else reviewing) to know why they are commented. Maybe you can add a comment above the whole section saying that the code section will be needed for a later PR. So that we know we do not need to actually review it, but simply skip it

Reply:
Removed (BTW, we have them on a fork), so that the code should be less 'messy'